projects/gigya/src/cms-components/gigya-js/gigya-js.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit, ChangeDetectorRef } from '@angular/core';
+import { BaseSiteService, LanguageService, ExternalJsFileLoader } from '@spartacus/core';
+import { GigyaConfig } from '../../config/gigya-config';
+
+declare var gigya: any;
+declare var window: Window;
+
+interface Window {
+  gigyaSpaLogin?: any;
+  GigyaRaasComponentReference?: any;
+  gigya?: any;
+  zone?: any;
+  __gigyaConf?: any;
+}
+
+window.__gigyaConf = { include: 'id_token' };
+
+window.gigyaSpaLogin = function (response) {
```

Comment:
Event listeners should all be handled in the raas component. Example solution for this: https://github.com/SAP/spartacus/pull/7824

I would avoid relying on window for anything that is not a gigya js or gigya configuration.
Better to use Angular mechanisms.

Reply:
Incorporated this change.